client_test.go

```diff
@@ -1,47 +1,184 @@
+		})
+	})
+
+	Convey("Given redis config with missing password", t, func() {
+		c, err := setUpClient("123.0.0.1", "", 0, 0)
+
+		Convey("Then the client will fail to be created", func() {
+			So(c, ShouldBeNil)
+			So(err, ShouldNotBeEmpty)
+			So(err.Error(), ShouldEqual, "password is missing")
+		})
+	})
+}
+
 func TestClient_Set(t *testing.T) {
-	Convey("Given redis client is set up", t, func() {
+	Convey("Given a valid session", t, func() {
+		mockRedisClient, mockClient := setUpMocks(*redis.NewStatusResult("success", nil), *redis.NewStringCmd())
 
-		mockClient := mock.ClientManagerMock{SetFunc: func(s *session.Session) error {
-			return nil
-		}}
+		s := &session.Session{
+			ID:           "1234",
+			Email:        "user@email.com",
+			Start:        time.Now(),
+			LastAccessed: time.Now(),
+		}
 
-		Convey("Where the session is valid", func() {
-			s := session.NewSession()
+		Convey("When cache attempts to store session", func() {
 			err := mockClient.Set(s)
 
-			Convey("Then the session should be Set in redis", func() {
-				So(mockClient.SetCalls(), ShouldHaveLength, 1)
+			Convey("Then session should be stored", func() {
 				So(err, ShouldBeNil)
+				So(mockRedisClient.SetCalls(), ShouldHaveLength, 1)
 			})
 		})
+	})
+
+	Convey("Given a valid session", t, func() {
+		mockRedisClient, mockClient := setUpMocks(*redis.NewStatusResult("fail", errors.New("failed to store session")), *redis.NewStringCmd())
+
+		s := &session.Session{
+			ID:           "1234",
+			Email:        "user@email.com",
+			Start:        time.Now(),
+			LastAccessed: time.Now(),
+		}
+
+		Convey("When cache attempts to store session", func() {
+			err := mockClient.Set(s)
 
+			Convey("Then session will not be stored", func() {
+				So(mockRedisClient.SetCalls(), ShouldHaveLength, 1)
+				So(err, ShouldNotBeEmpty)
+				So(err.Error(), ShouldEqual, "failed to store session")
+			})
+		})
 	})
 
-	Convey("Given redis client is set up", t, func() {
+	Convey("Given an invalid session", t, func() {
+		mockRedisClient, mockClient := setUpMocks(*redis.NewStatusCmd(), *redis.NewStringCmd())
 
-		mockClient := mock.ClientManagerMock{SetFunc: func(s *session.Session) error {
-			return errors.New("empty session provided")
-		}}
+		var s *session.Session = nil
 
-		Convey("Where the session is empty", func() {
-			var s *session.Session = nil
+		Convey("When cache attempts to store session", func() {
 			err := mockClient.Set(s)
 
-			Convey("Then session is not stored in redis", func() {
-				So(mockClient.SetCalls(), ShouldHaveLength, 1)
+			Convey("Then session will not be stored", func() {
+				So(mockRedisClient.SetCalls(), ShouldHaveLength, 0)
+				So(err, ShouldNotBeEmpty)
+				So(err.Error(), ShouldEqual, "session is empty")
+			})
+		})
+	})
+}
+
+func TestClient_GetByID(t *testing.T) {
+	Convey("Given a session ID", t, func() {
+		mockRedisClient, mockClient := setUpMocks(*redis.NewStatusCmd(), *redis.NewStringResult(string(resp), nil))
```

Comment:
I'm being picky name but `mockClient` is not a mock - its the target of this test. Rename it to `client` so its clear this is the struct being tested.

Reply:
Name has been changed.